src/vs/editor/browser/gpu/viewGpuContext.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { getActiveWindow } from '../../../base/browser/dom.js';
+import { createFastDomNode, type FastDomNode } from '../../../base/browser/fastDomNode.js';
+import { Emitter } from '../../../base/common/event.js';
+import { Disposable } from '../../../base/common/lifecycle.js';
+import { GPULifecycle } from './gpuDisposable.js';
+import { ensureNonNullable, observeDevicePixelDimensions } from './gpuUtils.js';
+
+export class ViewGpuContext extends Disposable {
+	readonly canvas: FastDomNode<HTMLCanvasElement>;
+	readonly ctx: GPUCanvasContext;
+
+	readonly device: Promise<GPUDevice>;
+
+	private readonly _onDidChangeCanvasDevicePixelDimensions = this._register(new Emitter<{ width: number; height: number }>());
+	readonly onDidChangeCanvasDevicePixelDimensions = this._onDidChangeCanvasDevicePixelDimensions.event;
```

Comment:
I think whenever you have such an event, it would be nice to also have the underlying current data available.
Otherwise it will motivate users of this event to not handle the initial (unchanged) case.

Reply:
I mean to have a getter for the current value, so it conceptually makes a complete observable value.

Reply:
The value is on `canvas` already. I had a try at actually exposing an observable this but got a little confused as to what the best thing to do here is (haven't the observable stuff much). I created https://github.com/microsoft/vscode/issues/228146 and we can quickly walk through this together tomorrow.